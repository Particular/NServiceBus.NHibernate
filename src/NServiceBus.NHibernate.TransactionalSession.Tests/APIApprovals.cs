namespace NServiceBus.TransactionalSession.Tests
{
    using NServiceBus.TransactionalSession;
    using NUnit.Framework;
    using Particular.Approvals;
    using PublicApiGenerator;

    [TestFixture]
    public class APIApprovals
    {
        [Test]
        public void Approve()
        {
            var publicApi = typeof(NHibernateOpenSessionOptions).Assembly.GeneratePublicApi(new ApiGeneratorOptions
            {
                ExcludeAttributes = ["System.Runtime.Versioning.TargetFrameworkAttribute", "System.Reflection.AssemblyMetadataAttribute"]
            });
            Approver.Verify(publicApi);
        }
    }
}
