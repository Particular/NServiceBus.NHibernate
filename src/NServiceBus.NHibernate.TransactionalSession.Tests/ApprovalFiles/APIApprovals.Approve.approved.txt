namespace NServiceBus.TransactionalSession
{
    public class NHibernateOpenSessionOptions : NServiceBus.TransactionalSession.OpenSessionOptions
    {
        public NHibernateOpenSessionOptions() { }
    }
    public static class NHibernateTransactionalSessionExtensions
    {
        public static NServiceBus.PersistenceExtensions<NServiceBus.NHibernatePersistence> EnableTransactionalSession(this NServiceBus.PersistenceExtensions<NServiceBus.NHibernatePersistence> persistenceExtensions) { }
        public static NServiceBus.PersistenceExtensions<NServiceBus.NHibernatePersistence> EnableTransactionalSession(this NServiceBus.PersistenceExtensions<NServiceBus.NHibernatePersistence> persistenceExtensions, NServiceBus.TransactionalSession.TransactionalSessionOptions transactionalSessionOptions) { }
        public static System.Threading.Tasks.Task Open(this NServiceBus.TransactionalSession.ITransactionalSession transactionalSession, System.Threading.CancellationToken cancellationToken = default) { }
    }
}